### ISSUE TITLE
Gate default state is {}, but i can create non-object Gate

### ISSUE THREAD

**Issue by ilyalesik** [zerobias/effector#119]
It's possible to create non-object Gate:
```js
const NumberGate = createGate<number>('number');
...
const Component = () => {
  useGate(NumberGate, 1)
}
```
and receive some runtime error after that, 
```js
NumberGate.state.watch((value) => {
  console.log(value * 2);
})
```
which will not be detected with type-checking, because `value` type is `number | null` and `value` can be `{}`.

## Relevant documentation

[createGate](https://effector.now.sh/en/api/effector-react/createGate)

**Comment by goodmind** [comment]
Fixed in 915dc1b1856513151269de89d480c9a419779eff

**Comment by ilyalesik** [comment]
🎉
